api4/channel.go

```diff
@@ -540,17 +540,20 @@ func deleteChannel(c *Context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	// Allow delete if user is the only member left in channel
-	if memberCount > 1 {
-		if channel.Type == model.CHANNEL_OPEN && !app.SessionHasPermissionToChannel(c.Session, channel.Id, model.PERMISSION_DELETE_PUBLIC_CHANNEL) {
-			c.SetPermissionError(model.PERMISSION_DELETE_PUBLIC_CHANNEL)
-			return
-		}
+	// Do not allow delete if there's only one member left in a private channel
```

Comment:
This appears to stop the last user in a private channel from deleting it, even if they have the permissions to via team/channel admin policy. Is this intended?